### PR TITLE
Improve support for react-router-redux

### PR DESCRIPTION
**Pull request by jhabdas**
**HELP PREVENT ADDITIONAL FORKS**

Users leveraging this module and using the last two versions of `react-redux-router` will experience breakage in their Segment reporting without this.

I cherry-picked the updates from a fork created by @thebigredgeek and revved on them. Here's an example showing the updated names coming from `react-redux-router` version `4.x`:

screen shot 2016-03-07 at 4 49 33 pm

### CODE REVIEW

**Comment by jhabdas** [comment]
Looks like I'll need to update some unit tests. Other than fixing what's breaking, should I add new tests as well to allow this PR in? I'd prefer not to have to maintain a fork just yet.

**Comment by jhabdas** [comment]
Adding tests for `react-router-redux` in next commit.

File: README.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ state is small and centralized and changes are explicit. You should,
 however, resist the temptation. This constraint is core to the **design
 philosophy** of Redux Segment.
 
-> Analytics is about **learning**  
+> Analytics is about **learning**
```
Comment (jhabdas) [inline README.md]: Hence the addition of the `.editorconfig` file.

File: src/index.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ function handleActionType(next: Function, action: Object) {
   switch (action.type) {
     case '@@router/INIT_PATH':
     case '@@router/UPDATE_PATH':
+    case '@@router/UPDATE_LOCATION':
```
Comment (jhabdas) [inline src/index.js]: Add routing action triggered when the `react-router-redux#4.0.0` [receives a location change](https://github.com/reactjs/react-router-redux/blob/v4.0.0/src/reducer.js#L5).

**Comment by bertrandk** [comment]
This is great. Thanks.

**Comment by jhabdas** [comment]
@bertrandk would you be opposed to falling forward to a newer version of `react-router` as I write my tests, like so in `package.json`: `"react-router": ">=1.0.3 < 3"`?

Doing so will help provide backwards compatibility for older router versions while allowing me to create new tests for `react-router-redux#4.0.0` (which depends on `react-router#2.0.0`, which is not currently supported) without making testing dependent on conditionally required dependencies (thereby keeping unit tests simple).

File: README.md (hunk 1 above)
Comment (bertrandk) [inline README.md]: Ha ha...fair enough. I honestly have no idea what editor I was using when writing that. My vim config will usually at least colour trailing whitespace as red, if not outright strip them on save.

**Comment by bertrandk** [comment]
@jhabdas sounds reasonable to me.

**Comment by jhabdas** [comment]
Muy bueno

File: package.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
   "description": "Segment.io analytics integration for redux.",
   "main": "redux-segment.js",
   "scripts": {
+    "lint": "eslint ./src",
```
Comment (jhabdas) [inline package.json]: Exposes ability to view new command `npm run` and ability to lint contents of `src` folder only with `npm run lint`.

File: package.json [hunk 4]
```diff
@@ -43,7 +44,8 @@
     "react-addons-test-utils": "^0.14.3",
     "react-dom": "^0.14.3",
     "react-redux": "^4.0.5",
-    "react-router": "^1.0.3",
+    "react-router": ">=1.0.3 < 3",
```
Comment (jhabdas) [inline package.json]: Don't force users to upgrade their routers, but allow users of `react-router#2.0.0` to get in on the fun.

**Comment by jhabdas** [comment]
@bertrandk could you give me Circle build with a clear cache on the following? https://circleci.com/gh/rangle/redux-segment/97

**Comment by jhabdas** [comment]
@bertrandk blocked pending Circle build with cache clear or confirmation the dependency archive currently in use under the box contains the updated dependency versions for `react-router#2.0.0` and `react-router-redux#4.0.0`.

**Comment by bertrandk** [comment]
Sorry for the delay @jhabdas...clean build in now in progress.

**Comment by jhabdas** [comment]
Thanks @bertrandk. I needed a sanity check and was having problems running tests locally so I'm hijacking your CI box while I get the tests to pass. Hope you don't mind!

**Comment by jhabdas** [comment]
@bertrandk Upon first run of `npm test` I observed dependencies download successfully (nice touch, by the way) and then the browser chrome opened and my tests ran with output. However, on subsequent runs the browser opened though became unresponsive - and test output in the console seems to have abended without producing output. Perhaps it'd be better to run tests headlessly. But this works for me.

**Comment by jhabdas** [comment]
Sidebar: I'm currently creating a way to spray data into AWS using Lambdas and this middleware. Fun stuff!

**Comment by jhabdas** [comment]
@bertrandk I totally just rewrote that last paragraph for clarity.

**Comment by jhabdas** [comment]
@bertrandk about to pick this back up. I saw `npm test` work again locally this morning, but only once (like yesterday). What I noticed this time is that the first opening of the browser the URL in the location bar is set to `http://localhost:49534/`. On subsequent runs it is blank.

**Comment by jhabdas** [comment]
@bertrandk Figured out what's happening. Looks like the test runner is spawning new instances of Chrome as opposed to opening new Windows. When the test window for a Chrome app instance is closed it leaves Chrome open. The workaround is to be sure to quit the new version of Chrome spawned. Otherwise tests cannot be run locally on subsequent runs.

**Comment by bertrandk** [comment]
@jhabdas Ah, that's a known issue: https://github.com/substack/browser-launcher/issues/34

Will address.

**Comment by jhabdas** [comment]
@bertrandk I've updated my last commit with the latest. I'm stuck on getting the new `react-redux-router` test to pass. When the new test is commented out all other tests pass. But when enable I have breakages in two tests, as described below.

With new test enabled, as in the commit pushed, I'm seeing following errors in both the `react-redux-router` and `redux-simple-router` router support tests, as shown here:

```
not ok 1 triggers page event on load
  ---
    operator: equal
    expected: 'page'
    actual:   undefined
  ...
not ok 2 triggers page event on navigation
  ---
    operator: equal
    expected: 'page'
    actual:   undefined
  ...
```

**Note:** With the update to `react-router@2` the `redux-simple-router` test throws warnings, which is expected given the latest major is [fully backwards compatible](https://github.com/reactjs/react-router/blob/master/upgrade-guides/v2.0.0.md#backwards-compatibility-and-deprecation-warnings).

**TL;DR**

I've tried several approaches:

- [Using a custom history](https://github.com/reactjs/react-router/blob/master/upgrade-guides/v2.0.0.md#using-custom-histories)
- Using both types of [`react-router-redux` history singletons](https://github.com/reactjs/react-router/blob/master/upgrade-guides/v2.0.0.md#history-singletons-provided).
- Even using a deprecated history object in the new test.

But I still haven't had any success. Perhaps you can help shed some light based on your experience with Tape and putting this very useful library together.

**Comment by jhabdas** [comment]
Sorry for the mouthful. Thoughts on switching out test runners or allowing this Router support affordance in without tests?

**Comment by bertrandk** [comment]
@jhabdas Taking a look. I'd expect to be able to cut a release with this fix in by tonight.

**Comment by jhabdas** [comment]
@bertrandk Thanks. Please let me know if there's anything I can do to help. We may end up creating another NPM module if we can't get this update in over the next week.

**Comment by jhabdas** [comment]
Removed new page test. @bertrandk ready for review, merge and bump if you're okay letting coverage levels slip a little. I spent several hours trying to get tests to pass unsuccessfully and would be more apt to create a new module with a different testing framework at this point just to move forward.

**Comment by bertrandk** [comment]
Yeah, I'm fine with that. I'm a little busier than I expected but I have a cleanup coming that should make this stuff much less painful. Thanks for bearing with things here @jhabdas. Pushing new version in the next 5 min.

**Comment by bertrandk** [comment]
Released in v0.7.1. Thanks again guys and sorry for the delay.

**Comment by jhabdas** [comment]
@bertrandk thanks for working this in. much appreciated
